### DISCUSSION TITLE
I would like to ask is there a possibility to keep keycloak.log file for more than one day?

### DISCUSSION THREAD

**Discussion by Mateusz-Krzyszpien** [keycloak/keycloak#24957] [Q&A]
Dears,

I have have basic configuration for logging into the file for our setup:

# Logging to file and directory path.
log=console,file
log-file=K:/\keycloak/\keycloak-21.0.1_with_sql/\log/\keycloak.log
log-level=INFO

Above properties are stored at keycloak.conf file.

Currently I can see that logs are stored only for one day, for example:
![image](https://github.com/keycloak/keycloak/assets/136449738/8a548653-e4b9-4d3d-a642-5c59e25ad8e0)

I would like to know how to change configuration so we could keep logs for month or even a year.

**Comment by danielFesenmeyer** [comment]
You could use quarkus-specific settings to increase rotation's `max-file-size` and/or `max-backup-index`, as described here: https://quarkus.io/guides/logging#quarkus-log-logging-log-config_quarkus.log.file-file-logging

Note that these configuration can NOT be specified in `keycloak.conf`. It can be defined in `quarkus.properties` in the same folder.

**Comment by Mateusz-Krzyszpien** [comment]
Dears,

I have set following properties at quarkus.properties:

quarkus.log.file.enable=true
quarkus.log.file.path=K:\keycloak\keycloak-21.0.1_with_sql\log\keycloak.log
quarkus.log.file.level=INFO
quarkus.log.file.rotation.max-backup-index=1000

Unfortunately file keycloak.log is not being created at directory that I specified. Files keycloak.conf and quarkus.properties are both at directory K:\keycloak\keycloak-21.0.1_with_sql\conf.

Do you know what else could be wrong?

**Reply by danielFesenmeyer** [reply]
Not sure, cause I'm no longer using Windows for coding. But in the quarkus config documentation I see a link to the java `File` javadoc: https://docs.oracle.com/javase/8/docs/api/java/io/File.html.

As far as I understand, your path should start with `K:\\`.

**Reply by Mateusz-Krzyszpien** [reply]
Dears,

I have tried with double backslashes:
quarkus.log.file.path=K:\\keycloak\keycloak-21.0.1_with_sql\log\keycloak.log

Unfortunately there is no improvement. Still logs are being not generated. 

I have came to conclusion that maybe my configuration from quarkus.properties file is not used by Keycloak at all. To test that theory I have removed quarkus.log.file.path property.
According to documentation it should create quarkus.log file somewhere. No such file is being created. Therefore I think, quarkus.properties are not  used by Keycloak anyhow.

Could you verify me theory and guide me further?
